src/pages/index.js

```diff
@@ -1,168 +1,241 @@
+      newCardSection.addItem(addCard);
+      addCardPopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      addCardPopup.renderLoading(false, "Create");
+    });
+}
+function handleAviPopup(inputValues) {
+  avatarPopup.renderLoading(true);
+  api
+    .updateProfileAvatar(inputValues.avatar)
+    .then(() => {
+      userInfo.setAvatar(inputValues.avatar);
+      avatarPopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      avatarPopup.renderLoading(false);
+    });
+}
+function handleDeleteClick(card) {
+  deleteCardPopup.open();
+  deleteCardPopup.setSubmitAction(() => {
+    deleteCardPopup.renderLoading(true);
+    api
+      .deleteCard(card.cardId)
+      .then(() => {
+        card.handleDeleteCard();
+        deleteCardPopup.close();
+      })
+
+      .catch((err) => {
+        console.log(err);
+      })
+      .finally(() => {
+        deleteCardPopup.renderLoading(false);
+      });
+  });
+}
 
 function handleProfileEditClick() {
   const user = userInfo.getUserInfo();
-
   userNameInput.value = user.name;
-
-  userDescriptionInput.value = user.description;
-
+  userDescriptionInput.value = user.about;
   profileEditPopup.open();
 }
 
-function handleProfileEditSubmit(inputValues) {
-  userInfo.setUserInfo(inputValues);
-  profileEditPopup.close();
+function handleLikeClick(card) {
+  if (card.isLiked) {
+    api
+      .removeCardLikes(card.cardId)
+      .then((res) => {
+        card.updateLikes(res.isLiked);
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  } else {
+    api
+      .addCardLikes(card.cardId)
+      .then((res) => {
+        card.updateLikes(res.isLiked);
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }
 }
 
-/* ----------------------------- add card modal ----------------------------- */
-
-const addCardButton = document.querySelector(".profile__add-button");
-
-const addCardPopup = new PopupWithForm("#card-edit-modal", handleAddCardSubmit);
-
-addCardPopup.setEventListeners();
-
-addCardButton.addEventListener("click", () => {
-  cardAddFormValidator.resetValidation;
-  cardAddFormValidator.toggleButtonState();
-  addCardPopup.open();
-});
-
-function handleAddCardSubmit(inputValues) {
-  const { title, link } = inputValues;
-  const newCardData = {
-    name: title,
-    link: link,
-  };
-  const newCard = createCard(newCardData);
-  section.addItem(newCard);
-  addCardPopup.close();
+function handleCardClick(cardData) {
+  previewImagePopup.open(cardData);
 }
 
 /* -------------------------------------------------------------------------- */
-
 /*                               FORM VALIDATION                              */
-
 /* -------------------------------------------------------------------------- */
-
+const aviFormValidator = new FormValidator(formValidatorConfig, aviEditForm);
+const editProfileFormValidator = new FormValidator(
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object `formValidators`.  And then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
const formValidators = {}

// enable validation
const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // here you get the name of the form
    const formName = formElement.getAttribute('name')

   // here you store a validator by the `name` of the form
    formValidators[formName] = validator;
   validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```